include/accessor.h

```diff
@@ -556,39 +571,26 @@ class accessor<DataT, Dims, Mode, target::host_task> : public detail::accessor_b
 
 template <typename DataT, int Dims = 1>
 class local_accessor {
-#if !CELERITY_FEATURE_LOCAL_ACCESSOR
-	static_assert(detail::constexpr_false<DataT>, "Your SYCL implementation cannot support celerity::local_accessor");
-#else
   private:
 #if WORKAROUND_DPCPP
 	using sycl_accessor = cl::sycl::accessor<DataT, Dims, cl::sycl::access::mode::read_write, cl::sycl::access::target::local>;
 #else
 	using sycl_accessor = cl::sycl::local_accessor<DataT, Dims>;
 #endif
 
+	template <typename Index>
+	using subscript_type = decltype(std::declval<sycl_accessor>()[std::declval<const Index&>()]);
```

Comment:
I feel like this is a bit far away from its only use (at line 623 as far as I can see).